server/src/main/scala/org/http4s/server/middleware/BracketRequestResponse.scala

```diff
@@ -203,4 +204,39 @@ object BracketRequestResponse {
     bracketRequestResponseCaseApp[F, A](acquire) { case (a, _) =>
       release(a)
     }
+
+  /** As [[#bracketRequestResponseRoutes]], but `acquire` and `release` are
+    * defined in terms of a [[cats.effect.Resource]].
+    *
+    * @note $releaseWarning
+    */
+  def bracketRequestResponseRoutesR[F[_], A](
+      resource: Resource[F, A]
+  )(implicit F: Bracket[F, Throwable]): ContextMiddleware[F, A] = {
+    // Compiler can't figure this out in 2.12, but it can in >= 2.13
+    val P: Profunctor[Kleisli[OptionT[F, *], *, *]] = Profunctor[Kleisli[OptionT[F, *], *, *]]
+    (contextRoutes: ContextRoutes[A, F]) =>
+      val contextRoutes0: ContextRoutes[(A, F[Unit]), F] =
+        P.lmap(contextRoutes)(_.map(_._1))
```

Comment:
If using `Profunctor` here is so painful - maybe just do `contextRoutes.local(_.map(_._1))`?

Reply:
@nigredo-tori sure. Updated.